File: code/718/Python/dp.py
class Solution:
    def findLength(self, A, B):
        m, n = len(A), len(B)
        dp = [[0]*n for _ in range(m)]
        ret = 0
        for i in range(m):
            if A[i]==B[0]:
                dp[i][0] = 1
                ret = max(ret, dp[i][0])
        for j in range(n):
            if A[0]==B[j]:
                dp[0][j] = 1
                ret = max(ret, dp[0][j])
        for i in range(1, m):
            for j in range(1, n):
                dp[i][j] = (dp[i-1][j-1] + 1) if A[i]==B[j] else 0
                ret = max(dp[i][j], ret)
        return ret

File: code/718/Java/dp.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int m=A.length, n=B.length;
        int[][] dp = new int[m][n];
        for (int i=0; i<m; i++)
            dp[i][0] = (A[i]==B[0]?1:0);
        for (int j=0; j<n; j++)
            dp[0][j] = (A[0]==B[j]?1:0);
        for (int i=1; i<m; i++)
            for (int j=1; j<n; j++)
                dp[i][j] = (A[i]==B[j]?(dp[i-1][j-1]+1):0);
        int ret = 0;
        for (int i=0; i<m; i++)
            for (int j=0; j<n; j++)
                ret = (dp[i][j]>ret)?dp[i][j]:ret;
        return ret;
    }
}